workbench

```diff
@@ -672,9 +672,505 @@ def add_media():
+        """Get the parent node ID of the media entity.
+
+        Parameters:
+            - get_media_response_body: The response body from a GET request to the media entity's endpoint.
+            - media_csv_row: The CSV row containing the media entity's field names and values.
+
+        Returns:
+            The parent node's ID if it corresponds to a valid node, otherwise None.
+
+        NOTE: If node_id is specified in the CSV row, that value will be returned. Otherwise, the first node ID in the list of nodes the media entity is attached to will be returned.
+        """
+        if 'node_id' in media_csv_row:  # If the CSV row contains a node ID, it takes precedence
+            if media_csv_row['node_id']:  # If the CSV row is not blank
+                return media_csv_row['node_id']
+        if not get_media_response_body['field_media_of']:  # If the media entity is not attached to any node
+            logging.error("Media ID %s is not attached to any node, which is a requirement for updating media files.", media_id)
+            return None
+        return get_media_response_body['field_media_of'][0]['target_id']  # Return the first node ID in the list of nodes the media entity is attached to
+
+    def patch_plain_text_fields(config: dict, media_type: str, media_csv_row: dict) -> dict:
+        """Create the JSON request to be sent to the media entity's PATCH endpoint for updating plain-text fields.
+
+        Parameters:
+            - config (dict): The global configuration object.
+            - media_type (str): The media entity's type (e.g., 'image').
+            - media_csv_row (dict): The CSV row containing the media entity's field names and values.
+
+        Returns:
+            - The JSON request (dict) to be sent to the media entity's PATCH endpoint.
+        """
+        media_json = {}
+        media_field_definitions = get_field_definitions(config, 'media', media_type)  # This will return a dict of field definitions keyed by field name.
+        for field_name, field_value in media_csv_row.items():  # Iterate through the CSV row's field names and values and find the fields that are plain-text.
+            if field_name in media_field_definitions:  # If the field name is a valid field name for the media entity's type
+                if 'string' in media_field_definitions[field_name]['field_type']:  # If field_name corresponds to a plain-text field. In Drupal 8, plain-text fields contain 'string' in their type.
+                    if field_value != '':  # If the field value is not empty
+                        media_json[field_name] = [{'value': field_value}]
+            else:
+                logging.warning("Field %s is not a valid plain-text field for media %s. This may be intended (e.g. media_id), but if not, please check the spelling of this field name.",
+                                field_name,
+                                media_type)
```

Comment:
`media_id` and `media_use_tid` are being logged here as invalid plain text field names. While true, I'd like to see those two fields not be logged, since it's a bit misleading to the user (since they are reserved column names) and also adds extraneous entries to the log file. Can we skip logging the warning for these two column names?

Reply:
Absolutely. I have committed the change now.

Reply:
Thanks! Smoke tested and works great!